kafka/producer_test.go

```diff
@@ -130,8 +130,13 @@ func TestNewProducerBasic(t *testing.T) {
 			fetches := client.PollRecords(nil, 1)
 			assert.Len(t, fetches.Records(), 0)
 
+			// kgo runs an unchecked goroutine to mark all the spans it creates as
+			// finished (and close the spans). So the only way to ensure it ran is to
+			// wait. Without this, we get flaky tests in CI.
+			time.Sleep(time.Millisecond)
```

Comment:
How about moving this into a loop with eg N=3 (there might be higher delays on CI) and starting with the `assert.Equal` check before the sleep?

Reply:
Good idea. Your comment actually reminded me that testify has an `assert.Eventually`, which is much better here.
https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually